infra/src/main/java/com/redhat/service/smartevents/infra/exceptions/BridgeErrorHelper.java

```diff
@@ -0,0 +1,78 @@
+package com.redhat.service.smartevents.infra.exceptions;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.platform.ProvisioningFailureException;
+
+import io.quarkus.runtime.Quarkus;
+
+@ApplicationScoped
+public class BridgeErrorHelper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BridgeErrorHelper.class);
+
+    private static final String USER_MESSAGE = "[%s] %s. Please contact support quoting UUID: %s";
+
+    private static final String USER_MESSAGE_UNKNOWN = "An error has occurred. Please contact support quoting UUID: %s";
+
+    @Inject
+    BridgeErrorService bridgeErrorService;
+
+    BridgeError deploymentFailedException;
+
+    @PostConstruct
+    protected void setup() {
+        Optional<BridgeError> error = bridgeErrorService.getError(ProvisioningFailureException.class);
+        if (error.isPresent()) {
+            deploymentFailedException = error.get();
+        } else {
+            LOGGER.error("{} error is not defined in the ErrorsService.", ProvisioningFailureException.class.getSimpleName());
+            Quarkus.asyncExit(1);
+        }
+    }
+
+    public BridgeErrorInstance getBridgeErrorInstance(Exception e) {
+        LOGGER.debug("Mapping '{}' to BridgeError", e.getClass().getName(), e);
+        return bridgeErrorService.getError(e.getClass())
+                .map(b -> {
+                    BridgeErrorInstance bei = new BridgeErrorInstance(b);
+                    LOGGER.info("Mapped '{}' to '{}'", e.getClass().getName(), bei);
+                    return bei;
+                })
+                .orElseGet(() -> {
+                    LOGGER.debug("'{}' not found in error catalog. Falling back to generic ProvisioningFailureException.", e.getClass().getName());
+                    BridgeErrorInstance bei = new BridgeErrorInstance(deploymentFailedException);
```

Comment:
Nitpick: Not sure if `ProvisioningFailureException` is generic enough. The `reason` of that exception looks good `An unspecified error occurred handling your request` but `Provisioning` sounds like something that can occur only during the provisioning. I'm wondering if `InternalPlatformException` is better in this case. In any case up to you, just brainstorming

Reply:
Fair point. Removed class and used `InternalPlatformException` in its place.